desk/lib/test-negotiate-agent.hoon

```diff
@@ -0,0 +1,159 @@
+::  test-agent-negotiate: like /lib/test-agent, but negotiation transparency
+::
+::    use this as you would /lib/test-agent. this library will hide most
+::    /lib/negotiate behavior and side-effects, letting you write tests against
+::    your agent's logic without having to simulate version negotation.
+::
+::    caveats:
+::    - the bowl will still contain /~/negotiate wires
+::
+/+  *test-agent, negotiate
+::
+|%
++$  card  card:agent:gall
+::TODO  +get-bowl
+++  do
+  |=  call=$-(state [(list card) agent:gall])
+  =/  m  (mare ,(list card))
+  ::NOTE  ^do will play the cards with their negotiate wires
+  ;<  caz=(list card)  bind:m  (^do call)
+  ::  simulate immediate matching versions for everywhere we negotiate
+  ::
+  ;<  res=cage  bind:m  (got-peek /x/~/negotiate/config)
+  =+  !<(=config:negotiate q.res)
+  =|  cards=(list card)
+  |-  =*  loop  $
+  ?^  caz
+    ?.  ?=([%pass [%~.~ %negotiate %heed @ @ @ ~] *] i.caz)
+      loop(caz t.caz, cards (snoc cards i.caz))
+    =*  who  (slav %p i.t.t.t.p.i.caz)
+    =*  dud         i.t.t.t.t.p.i.caz
+    =*  pro       i.t.t.t.t.t.p.i.caz
+    ::NOTE  at time of writing, /lib/negotiate no-ops on watch acks,
+    ::      so we can get away with not processing the cards here.
+    ;<  *  bind:m  (^do-agent p.i.caz [who dud] %watch-ack ~)
+    ;<  c=(list card)  bind:m
+      %^  do-agent  p.i.caz
+        [who dud]
+      :+  %fact  %noun
+      !>((~(got by (~(got by config) dud)) pro))
+    loop(caz t.caz, cards (weld cards c))
+  ::  filter out /lib/negotiate internal effects,
+  ::  and clean up wrapped wires
+  ::
+  %-  pure:m
+  %+  murn  cards
+  |=  =card
+  ^-  (unit ^card)
+  ?+  card  `card
+      [%pass [%~.~ %negotiate %inner-watch @ @ *] *]
+    `card(p t.t.t.t.t.p.card)
```

Comment:
I mean, this works, but what about saying "skip 5 elements". Hopefully this typechecks :-)
```suggestion
    `card(p |5.p.card)
```